rosbag2_storage/include/rosbag2_storage/serialized_bag_message.hpp

```diff
@@ -27,7 +27,15 @@ namespace rosbag2_storage
 struct SerializedBagMessage
 {
   std::shared_ptr<rcutils_uint8_array_t> serialized_data;
-  rcutils_time_point_value_t time_stamp;
+  /**
+   * @brief Nanosecond timestamp when this message was received.
+   */
+  rcutils_time_point_value_t receive_time_stamp;
+  /**
+   * @brief Nanosecond timestamp when this message was initially published. If
+   * not available, this will be set to time_stamp.
+   */
+  rcutils_time_point_value_t publish_time_stamp;
```

Comment:
@jmachowinski Could you please rename variables with timestamps?
```suggestion
  /**
   * @brief Nanosecond timestamp when this message was received.
   */
  rcutils_time_point_value_t recv_timestamp;
  /**
   * @brief Nanosecond timestamp when this message was initially published. If
   * not available, this will be set to recv_timestamp.
   */
  rcutils_time_point_value_t send_timestamp;
```
The motivation is following:
1. Don't use underscore between `time_stamp` 
  - a) for consistency with rclcpp and other core packages. it will be shorter in writing and it looks and reads like one entity `timestamp`. 
  - b) We usually divide with an underscore when more than one logical entity is combined. I know in some places in rosbag2 we are already using `time_stamp` but let's not repeat that style and make it better.
2. The `recv` and `send` is shorter than `receive` and `publish`.
3. The `publish` might be confusing. Since when one will read its name it will be not clear whether this timestamp for the player with what timestamp this message will be published during playback or some other sophisticated logic or truly timestamp when the message was published. It would be more clear if it would be `received_timestamp` and `published_timestamp`. However,  `recv_timestamp` and `send_timestamp` are shorter and also not introducing confusion about `publish`.